### ISSUE TITLE
Freedraw lines endings turn sharply despite minimal deltas

### ISSUE THREAD

**Issue by dwelle** [excalidraw/excalidraw#3574]
(I recall we've talked about this or something similar on chat.)

Often when I draw relatively straight lines the endings end in a sharp turn or a pointy end. I might move the cursor slightly on pointerup.

![image](https://user-images.githubusercontent.com/5153846/118160160-3d931500-b41e-11eb-8791-1c93d70fa5cd.png)

This one's ending turns a lot despite the last two points having same coords:

![image](https://user-images.githubusercontent.com/5153846/118160011-07559580-b41e-11eb-8570-0a784a23524c.png)

```
"points": [
  [0, 0],
  [1, 5],
  [4, 10],
  [8, 18],
  [11, 25],
  [15, 31],
  [18, 33],
  [20, 33],
  [21, 33],
  [21, 31],
  [21, 31]
],
```

Or consider these lines:

![image](https://user-images.githubusercontent.com/5153846/118159680-9d3cf080-b41d-11eb-948e-5345a71d4b73.png)

Both look very different, and the B's ending turns a lot despite the points of B differing in just the last `[14, 27]`, where `x` coord is just `1` less from prev point:

```
// A
[0, 0],
[1, 2],
[4, 6],
[11, 17],
[15, 22],
[16, 26],
[16, 27],
[15, 27]
```

```
[0, 0],
[1, 2],
[4, 6],
[11, 17],
[15, 22],
[16, 26],
[16, 27],
[15, 27],
[14, 27]
```

source: https://excalidraw.com/#json=6705652302872576,SamJrFGInpDkDXe6dtAZ9w

/cc @steveruizok

**Comment by steveruizok** [comment]
This is actually such a frustrating problem. 😰 So far I haven't found a solution that does not make some other (more noticeable) part of the algorithm worse.

The question comes down to how best to handle end caps: when to discard points at the end if they include a short curve, or whether to align points at the end (which makes corners behave strangely while drawing), or whether to have a smarter way of discarding cap points based on some intersection.

**Comment by dwelle** [comment]
Yeah, I imagine this is not going to be easy indeed. But thought we should track it anyway.
